mac/utils/utils.c

```diff
@@ -0,0 +1,66 @@
+//
+//  utils.c
+//  xdag
+//
+//  Created by Rui Xie on 3/16/18.
+//  Copyright © 2018 xrdavies. All rights reserved.
+//
+
+#include "utils.h"
+#if defined (__APPLE__)|| defined (__MACOS__)
+#include <unistd.h>
+#include <libgen.h>
+#define PATH_MAX 4096
+#elif _WIN32
+#include <shlwapi.h>
+#else
+#include <libgen.h>
+#include <linux/limits.h>
+#include <unistd.h>
+#endif
+
+static char g_xdag_current_path[4096] = {0};
+
+void xdag_init_path(char *path) {
+#if defined (__APPLE__)|| defined (__MACOS__)
+    char * n = dirname(path);
+    if (*n != '/' && *n != '\\') {
+        char buf[PATH_MAX];
+        getcwd(buf, PATH_MAX);
+        sprintf(g_xdag_current_path, "%s/%s", buf, n);
+    } else {
+        sprintf(g_xdag_current_path, "%s", n);
+    }
+    
+#elif _WIN32
+    char szPath[MAX_PATH];
+    char szBuffer[MAX_PATH];
+    char * pszFile;
+    
+    GetModuleFileName(NULL, (LPTCH)szPath, sizeof(szPath) / sizeof(*szPath));
+    GetFullPathName((LPTSTR)szPath, sizeof(szBuffer) / sizeof(*szBuffer), (LPTSTR)szBuffer, (LPTSTR*)&pszFile);
+    *pszFile = 0;
+    
+    sprintf(g_xdag_current_path, "%s", szBuffer);
```

Comment:
![default](https://user-images.githubusercontent.com/6870809/37542280-b223a58a-296e-11e8-91dd-b1977c41d58a.png)
Runtime error. I will try to resolve it, since it is Windows section.

Reply:
Fixed. There was error with project settings o_O.